helix-term/src/handlers/completion.rs

```diff
@@ -291,6 +300,213 @@ fn request_completion(
     });
 }
 
+fn path_completion(
+    cursor: usize,
+    text: helix_core::Rope,
+    doc: &helix_view::Document,
+) -> Option<BoxFuture<'static, anyhow::Result<Vec<CompletionItem>>>> {
+    if !doc.supports_path_completion() {
+        return None;
+    }
+
+    // TODO find a good regex for most use cases (especially Windows, which is not yet covered...)
+    // currently only one path match per line is possible in unix
+    static PATH_REGEX: Lazy<rope::Regex> =
+        Lazy::new(|| rope::Regex::new(r"((?:~|\$HOME|\$\{HOME\})?(?:\.{0,2}/)+.*)").unwrap());
+
+    let cur_line = text.char_to_line(cursor);
+    let start = text.line_to_char(cur_line).max(cursor.saturating_sub(1000));
+    let line_until_cursor = text.slice(..).regex_input_at(start..cursor);
+
+    let (dir_path, typed_file_name) = PATH_REGEX
+        .find_iter(line_until_cursor)
+        .last() // this is a workaround for missing "match end of slice" support in `rope::Regex`
```

Comment:
regex cursor works the same way as regex automata with regards to anchors/search ranges. What you need to do is anchor you regex (put `$` at the end) and instead of using `text.slice(...).regex_input_at(start..cursor` us `text.slice(start..cursor).regex_input_at(..)`. 

The difference is that `regex_input_at(start..end)` runs a regex within a part of a string/rope but still takes the surrounding context into account for lookaround assertions (so `\b` works correctly and so `^` only matches start/end of entire string and not the start of the search range).